pal/src/map/common.cpp

```diff
@@ -49,7 +49,7 @@ INT W32toUnixAccessControl( IN DWORD flProtect )
         MemAccessControl = PROT_EXEC | PROT_READ | PROT_WRITE;
         break;
     case PAGE_EXECUTE :
-        MemAccessControl = PROT_EXEC;
+        MemAccessControl = PROT_EXEC | PROT_READ; // WinAPI PAGE_EXECUTE also implies readable
```

Comment:
Update the PR to replace PAGE_EXECUTE with PAGE_EXECUTE_READ when appropriate, while still keeping PAL matches Windows, this change makes the code easier to read and causes less confusion since we use PAGE_EXECUTE_READ explicitly instead of using the implicit meaning of PAGE_EXECUTE when we actually read from it. 

@curtisman @obastemur PTAL